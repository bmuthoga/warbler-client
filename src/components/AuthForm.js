import React, { Component } from 'react';

class AuthForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      username: '',
      password: '',
      profileImageUrl: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit(e) {
    e.preventDefault()
    const authType = this.props.signUp ? 'signup' : 'signin'
    this.props.onAuth(authType, this.state)
      .then(() => {
        this.props.history.push('/')
      })
      .catch(() => {
        return
      })
  }

  render() {
    const { 
      email, 
      username, 
      password, 
      profileImageUrl 
    } = this.state
    const { 
      buttonText, 
      heading, 
      signUp, 
      errors, 
      history, 
      removeError 
    } = this.props

    if (errors.message) {
      const unlisten = history.listen(() => {
        removeError()
        unlisten()
      })
    }

    return (
      <div>
        <div className="row justify-content-md-center text-center">
          <div className="col-md-6">
            <form onSubmit={this.handleSubmit}>
              <h2>{heading}</h2>
              {errors.message &&
                <div className='alert alert-danger'>
                  {errors.message}
                </div>
              }
              <label htmlFor='email'>Email:</label>
              <input
                type='text'
                className='form-control'
                id='email'
                name='email'
                value={email}
                onChange={this.handleChange}
              />
              <label htmlFor='password'>Password:</label>
              <input
                type='password'
                className='form-control'
                id='password'
                name='password'
                value={password}
                onChange={this.handleChange}
              />
              {signUp && (
                <div>
                  <label htmlFor='username'>Username:</label>
                  <input
                    type='text'
                    className='form-control'
                    id='username'
                    name='username'
                    value={username}
                    onChange={this.handleChange}
                  />
                  <label htmlFor='image-url'>Image URL:</label>
                  <input
                    type='text'
                    className='form-control'
                    id='image-url'
                    name='profileImageUrl'
                    value={profileImageUrl}
                    onChange={this.handleChange}
                  />
                </div>
              )}
              <button type='submit' className='btn btn-primary btn-block btn-lg'>
                {buttonText}
              </button>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

export default AuthForm;
