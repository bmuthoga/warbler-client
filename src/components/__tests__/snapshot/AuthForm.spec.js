import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import AuthForm from '../../AuthForm'

const errors = { message: null }

it('should render Login form correctly', () => {
  const wrapper = shallow(
    <AuthForm
      buttonText='Log in'
      heading='Welcome Back'
      errors={errors}
      removeError={jest.fn()}
    />
  )

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('should render SignUp form correctly', () => {
  const wrapper = shallow(
    <AuthForm
      buttonText='Sign me up!'
      heading='Join Warbler Today'
      onAuth={jest.fn()}
      errors={errors}
      removeError={jest.fn()}
      signUp
    />
  )

  expect(toJson(wrapper)).toMatchSnapshot()
})
