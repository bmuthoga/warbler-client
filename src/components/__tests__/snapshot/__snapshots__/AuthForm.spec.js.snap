// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm Snaphshot Tests should render Login form correctly 1`] = `
<div>
  <div
    className="row justify-content-md-center text-center"
  >
    <div
      className="col-md-6"
    >
      <form
        onSubmit={[Function]}
      >
        <h2>
          Welcome Back
        </h2>
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <button
          className="btn btn-primary btn-block btn-lg"
          type="submit"
        >
          Log in
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`AuthForm Snaphshot Tests should render SignUp form correctly 1`] = `
<div>
  <div
    className="row justify-content-md-center text-center"
  >
    <div
      className="col-md-6"
    >
      <form
        onSubmit={[Function]}
      >
        <h2>
          Join Warbler Today
        </h2>
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <div>
          <label
            htmlFor="username"
          >
            Username:
          </label>
          <input
            className="form-control"
            id="username"
            name="username"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            htmlFor="image-url"
          >
            Image URL:
          </label>
          <input
            className="form-control"
            id="image-url"
            name="profileImageUrl"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <button
          className="btn btn-primary btn-block btn-lg"
          type="submit"
        >
          Sign me up!
        </button>
      </form>
    </div>
  </div>
</div>
`;
